ldap/servers/slapd/back-ldbm/db-mdb/mdb_import_threads.c

```diff
@@ -900,24 +910,46 @@ dbmdb_import_entry_info_by_param(EntryInfoParam_t *param, WorkerQueueData_t *wqe
         }
     }
     if (DNRC_IS_ENTRY(dnrc)) {
-        /* Note: If someday we needs to rebuild the entry dn, we could store it in the entry info
-         *  and build it from parent entry info + entry rdn. But as of today we do not need it.
-         */
-        len = strlen(nrdn) + strlen(rdn) + 2 + (5 + pinfo[1]) * sizeof(ID);
+        size_t rdnlen = strlen(rdn);
+        size_t dnlen = 0;
+        if (param->flags & EIP_RDN) {
+            /* In reindex case, dn must be rebuilt to be able to scope properly the vlv index */
+            dnlen = rdnlen + 1 + pinfo[INFO_IDX_DN_LEN];  /* dn len (including final \0) */
+        }
+
+        len = rdnlen + strlen(nrdn) + 2 + dnlen + (INFO_IDX_ANCESTORS + 1 + pinfo[INFO_IDX_NB_ANCESTORS]) * sizeof(ID);
```

Comment:
I agree: it is cleaner - I added rdnlen because I needed it to compute the dn size but I did not need to use nrdnlen for the dn size computation so I just did not think about it ...